packages/bpk-component-segmented-control/src/BpkSegmentedControl.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ReactNode} from 'react';
+import { useState } from 'react';
+
+import { cssModules } from '../../bpk-react-utils';
+
+import SEGMENT_TYPES from './segmentTypes';
+
+import type { SegmentTypes } from './segmentTypes';
+
+import STYLES from './BpkSegmentedControl.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export type Props = {
+  buttonContents: string[] |  ReactNode[];
```

Comment:
Only difference between the two potential Segmented Controls are that one expects an Array of strings, the other an array of objects. 

I do not see a world where there will be a mixture of strings and elements, but wanted to check if you think the same? Is there a preference to have an Array<stings | elements> ?
<img width="827" alt="Screenshot 2024-06-06 at 15 06 16" src="https://github.com/Skyscanner/backpack/assets/86477000/a581e0e6-2437-419c-807f-b9c89cbbe539">

Reply:
Yeah we would never have a mix of them at the same time, so would only exist as separate versions